apps/service_providers/llm_service/state.py

```diff
@@ -355,3 +331,68 @@ def citations_enabled(self):
     @property
     def assistant(self):
         return self.experiment.assistant
+
+
+class PipelineAssistantState(BaseAssistantState):
+    def __init__(
+        self,
+        assistant: OpenAiAssistant,
+        session: ExperimentSession,
+        trace_service=None,
+        input_formatter: str | None = None,
+        citations_enabled: bool = False,
+    ):
+        self._assistant = assistant
+        self.session = session
+        self.trace_service = trace_service
+
+        self.input_formatter = input_formatter
+        self._citations_enabled = citations_enabled
+        self.input_message_metadata = {}
+        self.output_message_metadata = {}
+
+    @property
+    def chat(self):
+        return self.session.chat
+
+    @cache
+    def get_llm_service(self):
+        return self.assistant.llm_provider.get_llm_service()
+
+    def _get_provider_model_name(self) -> str:
+        return self.assistant.llm_provider_model.name
+
+    def get_chat_model(self):
```

Comment:
Where is this called from?

Reply:
Hmm, looks like another copy pasta leftover. I copied a bunch of methods in one go and didn't double check this method's usage. Should be fixed [here](https://github.com/dimagi/open-chat-studio/pull/869/commits/b64af838631ec1744469ff955cb828d13f28a114)